buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes.groovy

```diff
@@ -163,6 +163,9 @@ timeout(time: TIMEOUT_TIME.toInteger(), unit: TIMEOUT_UNITS) {
                                         cleanDirsStr += cleanDirs.join(" ${buildWorkspace}/../../")
```

Comment:
This, or `cleanDirs` has to change; as is, this still includes `${buildWorkspace}/../../test*` which will remove the `testDependency` directory.

Reply:
@keithc-ca  `cleanDirsStr = cleanDirsStr.replaceAll("${buildWorkspace}/../../test\\*", "")` this line would exclude removing test* items from `${buildWorkspace}/../../ ` directory.

Reply:
Yes, I missed that line.

Reply:
I suggest a better approach would be to add `'test*'` to `cleanDirs` only for non-Windows nodes.

Reply:
@keithc-ca ,
I think even in windows we are required to clean test* dirs in /tmp directory. The only exclusion is for the testDependency dir. Is that not right @llxia @AdamBrousseau?

Reply:
Right, the only exclusion/special case is, because we clean HOME on Windows, we need to **not** remove `$HOME/testDependency` dir as that is a cache. I believe there are dirs or files in /tmp with the pattern `test*`

Reply:
Another thought occurred to me, but I don't imagine the consequences are pleasant: what if we change the name `testDependency` to something else (that doesn't start with "test")?

Reply:
That would be up to @llxia . Is this not easy to solve here?

Reply:
If the testDependency is renamed to something else that would also be a fix for the situation if at all there is a probability its name might get changed in the future.
cc @keithc-ca , @AdamBrousseau , @llxia

Reply:
It looks to me like the name `testDependency` is only used in [JenkinsfileBase](https://github.com/adoptium/aqa-tests/blob/2e34bbfe9931be699284120c04acf9a42680f71e/buildenv/jenkins/JenkinsfileBase#L108-L109) to define `LIB_DIR` which is used everywhere else. We might get away with changing the name in that one file.

Reply:
Yes, testDependency dir is set at JenkinsfileBase. If we rename it, the corresponding [clean-up job](https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Cleanup_Test_Lib/) should be updated as well.

Reply:
So, are we agreed that we should change those jobs to use a different name than `testDependency`, say to `external` or `support`?

Reply:
@llxia what should be the action taken?

Reply:
@smlambert  we are going to change `testDependency` to `externalDependency` at https://github.com/adoptium/aqa-tests/blob/2e34bbfe9931be699284120c04acf9a42680f71e/buildenv/jenkins/JenkinsfileBase#L108-L109. Do you have any objections?

FYI @annaibm @LongyuZhang

Reply:
Thanks @llxia.  It will be good to raise an aqa-tests issue for awareness for others that the folder name will be changed to better support automatic cleaning of test material, while maintaining the benefit of preloading test dependencies.  Other than communicating the change and checking that any references to that folder are accounted for, I have no objections.